docs/extension-development/integrating-coming-soon-mode.md

```diff
@@ -0,0 +1,77 @@
+---
+post_title: Integrating with coming soon mode
+tags: how-to, coming-soon
+---
+
+# Integrating with coming soon mode
+
+This guide provides examples for third-party developers and hosting providers on how to integrate their systems with WooCommerce's coming soon mode. For more details, please read the [developer blog post](https://developer.woocommerce.com/2024/06/18/introducing-coming-soon-mode/).
+
+## Clear server cache on site visibility settings change
+
+When the site's visibility settings is changed, it may be necessary to clear a server cache in order for the changes to be applied and customer-facing pages re-cached. We can utilize [`update_option`](https://developer.wordpress.org/reference/hooks/update_option/) hook to achieve this.
+
+```php
+add_action( 'update_option_woocommerce_coming_soon', 'clear_server_cache', 10, 3 );
+add_action( 'update_option_woocommerce_store_pages_only', 'clear_server_cache', 10, 3 );
+
+function clear_server_cache( $old_value, $new_value, $option ) {
+    // Implement your logic to clear the server cache.
+    if ( function_exists( 'your_cache_clear_function' ) ) {
+        your_cache_clear_function();
+    }
+}
+```
+
+## Syncing coming soon mode with other plugins
+
+You can programmatically sync the coming soon mode from your plugin or application. Here are some example use cases:
+
+- Integrating with a maintenance mode plugin.
+- Integrating with hosting provider's coming soon mode.
+
+### Trigger from WooCommerce
+
+```php
+// Priority set to 11 to ensure the options are already saved.
+add_action( 'woocommerce_update_options_site-visibility', 'sync_coming_soon_to_other_plugins', 11 );
```

Comment:
I think this will fire when the user sets coming soon mode from site visibility settings, but we will need to account for direct changes to the options 🤔 

For example, I think the LYS hub does an `update_option` call to launch the store. We can hook directly like this:
```
add_action( 'update_option_woocommerce_coming_soon', 'my_update', 10, 3 );

function my_update( $old_value, $new_value, $option ) {
	$is_enabled = $new_value === 'yes';

    // Implement your logic to sync coming soon status.
    if ( function_exists( 'set_your_plugin_status' ) ) {
        set_your_plugin_status( $is_enabled );
    }
}
```

Reply:
@adrianduffell  Good catch, my approach will not handle direct option change via lys task. However, I have a concern for a 2-way sync.

For example, from external plugin, we call `update_option`, which in turn will run the `update_option_woocommerce_coming_soon ` callback, which will then call `sync_coming_soon_to_other_plugins`, so the external plugin might set itself unintentionally in a loop.

```mermaid
graph TD;
    external_plugin-->update_option_woocommerce_coming_soon;
    update_option_woocommerce_coming_soon-->my_update;
    my_update-->sync_coming_soon_to_other_plugins;
    sync_coming_soon_to_other_plugins-->external_plugin;
```

A solution to this might be to disable the action when we update it (ChatGPT suggestion). What do you think?

```php
function sync_coming_soon_from_other_plugins( $is_enabled ) {
    // Temporarily remove the action to prevent an infinite loop.
    remove_action( 'update_option_woocommerce_coming_soon', 'my_update', 10, 3 );

    // Set coming soon mode.
    if ( isset( $is_enabled ) ) {
        update_option( 'woocommerce_coming_soon', $is_enabled ? 'yes' : 'no' );
    }

    // Re-add the action.
    add_action( 'update_option_woocommerce_coming_soon', 'my_update', 10, 3 );
}
```

Reply:
Oof, yeah that infinite loop is a real concern 😅 Disabling the action looks good for now, but I suppose this is a sign we should work on designing a safer interface for hosts to keep in sync.

Nice chart btw! ❤️

Reply:
@adrianduffell I've fixed the `Trigger from WooCommerce` with your suggestion and added example for 2-way sync in 39f8722a59119845d1a2b8632b341fbbbd90ba12. Let me know what you think!

Reply:
@adrianduffell I've also updated the docs with your suggestion on following the style guide. Please have another look!